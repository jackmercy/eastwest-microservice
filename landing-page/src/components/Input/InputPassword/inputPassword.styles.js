import { createUseStyles } from 'react-jss';

import { rem } from '../../../utils/jss';
import BREAKPOINTS from '../../../constants/BREAKPOINTS';

export default createUseStyles({
  inputWrapper: {
    position: 'relative',
    '& > [class^=icon-]': {
      top: '50%',
      right: rem(16),
      fontSize: rem(20),
      cursor: 'pointer',
      position: 'absolute',
      transform: 'translateY(-50%)',
    },
  },
  input: { padding: [rem(11), rem(40), rem(11), rem(16)] },

  [`@media only screen and (max-width: ${BREAKPOINTS.maxSm}px)`]: {
    inputWrapper: {
      '& > [class^=icon-]': { right: rem(12), fontSize: rem(16) },
    },
    input: { padding: [rem(3), rem(32), rem(3), rem(16)] },
  },
});
