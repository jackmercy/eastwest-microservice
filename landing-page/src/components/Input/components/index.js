export { default as FieldLabel } from './FieldLabel';
export { default as FieldError } from './FieldError';
export { default as FieldInput } from './FieldInput';
