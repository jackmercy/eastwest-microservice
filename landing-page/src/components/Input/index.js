export * from './components';

export { default } from './Input';
export { default as InputNumber } from './InputNumber';
export { default as InputPhone } from './InputPhone';
export { default as InputSearch } from './InputSearch';
export { default as InputPassword } from './InputPassword';
