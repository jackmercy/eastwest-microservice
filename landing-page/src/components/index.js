export { default as SEO } from './SEO';
export { default as Logo } from './Logo';
export { default as Modal } from './Modal';
export { default as Input } from './Input';
export { default as Select } from './Select';
export { default as Button } from './Button';
export { default as Loading } from './Loading';
export { default as Carousel } from './Carousel';
export { default as Textarea } from './Textarea';
export { default as CheckBox } from './CheckBox';
export { default as PopupMenu } from './PopupMenu';
export { default as StaticHTML } from './StaticHTML';

export * from './Input';
export * from './Modal';
