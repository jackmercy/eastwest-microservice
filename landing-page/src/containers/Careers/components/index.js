export { default as SectionTitle } from './SectionTitle';
export { default as MenuScrollableLayout } from './MenuScrollableLayout';
