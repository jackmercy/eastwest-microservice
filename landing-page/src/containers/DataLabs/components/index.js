export { default as Teaser } from './Teaser';
export { default as JoinUs } from './JoinUs';
export { default as Become } from './Become';
export { default as HowItWorks } from './HowItWorks';
export { default as ReadyToStart } from './ReadyToStart';
