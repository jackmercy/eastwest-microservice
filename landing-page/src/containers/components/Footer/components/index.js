export { default as Register } from './Register';
export { default as CopyRight } from './CopyRight';
export { default as SummaryInfo } from './SummaryInfo';
export { default as SectionTitle } from './SectionTitle';
export { default as SwitchLanguages } from './SwitchLanguages';
