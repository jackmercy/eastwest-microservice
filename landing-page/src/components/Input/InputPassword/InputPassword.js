import React, { memo, useState } from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

import { FieldLabel, FieldInput, FieldError } from '../components';

import useStyles from './inputPassword.styles';

const InputPassword = ({
  name,
  label,
  error,
  dataTest,
  required,
  className,
  inputClassName,
  labelClassName,
  errorClassName,
  ...restProps
}) => {
  const [showValue, setValueStatus] = useState();
  const classes = useStyles();

  const toggleValueStatus = () => {
    setValueStatus(!showValue);
  };

  return (
    <div
      className={className}
      data-test={dataTest || 'component-input-password-root'}
    >
      {label ? (
        <FieldLabel
          label={label}
          htmlFor={name}
          required={required}
          className={labelClassName}
        />
      ) : null}
      <div className={classes.inputWrapper}>
        <FieldInput
          name={name}
          hasError={!!error}
          required={required}
          className={classNames(classes.input, inputClassName)}
          {...restProps}
          type={showValue ? 'text' : 'password'}
        />
        <i
          role="presentation"
          className="icon-eye"
          onMouseDown={toggleValueStatus}
          onMouseUp={toggleValueStatus}
          onTouchStart={toggleValueStatus}
          onTouchEnd={toggleValueStatus}
          data-test="component-input-password-icon"
        />
      </div>
      {error ? <FieldError error={error} className={errorClassName} /> : null}
    </div>
  );
};

InputPassword.propTypes = {
  name: PropTypes.string.isRequired,
  label: PropTypes.node,
  error: PropTypes.string,
  required: PropTypes.bool,
  dataTest: PropTypes.string,
  className: PropTypes.string,
  inputClassName: PropTypes.string,
  labelClassName: PropTypes.string,
  errorClassName: PropTypes.string,
};
InputPassword.defaultProps = {
  error: '',
  dataTest: '',
  className: '',
  required: false,
  label: undefined,
  inputClassName: '',
  labelClassName: '',
  errorClassName: '',
};

export default memo(InputPassword);
