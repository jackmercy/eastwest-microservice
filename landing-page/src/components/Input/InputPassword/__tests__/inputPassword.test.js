import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { FieldLabel, FieldInput, FieldError } from '../../components';
import InputPassword from '../InputPassword';

describe('Testing Input component > InputPassword', () => {
  let wrapper;
  const props = { label: 'Password', name: 'password', error: 'some error' };

  beforeAll(() => {
    wrapper = mount(<InputPassword {...props} />);
  });

  it('should render match snapshot', () => {
    const wrapper = shallow(<InputPassword {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should have a FieldLabel', () => {
    expect(wrapper.find(FieldLabel)).toHaveLength(1);
  });

  it('should have a FieldInput', () => {
    expect(wrapper.find(FieldInput)).toHaveLength(1);
  });

  it('should have a eye icon', () => {
    expect(
      wrapper.find('[data-test="component-input-password-icon"]'),
    ).toHaveLength(1);
  });

  it('should have a FieldError', () => {
    expect(wrapper.find(FieldError)).toHaveLength(1);
  });
});
