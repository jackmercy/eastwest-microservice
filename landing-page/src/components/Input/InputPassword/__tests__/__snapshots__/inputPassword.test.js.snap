// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Input component > InputPassword should render match snapshot 1`] = `
<div
  className=""
  data-test="component-input-password-root"
>
  <Memo(FieldLabel)
    className=""
    htmlFor="password"
    label="Password"
    required={false}
  />
  <div
    className="inputWrapper-0-2-1"
  >
    <Memo(FieldInput)
      className="input-0-2-2"
      hasError={true}
      name="password"
      required={false}
      type="password"
    />
    <i
      className="icon-eye"
      data-test="component-input-password-icon"
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      role="presentation"
    />
  </div>
  <Memo(FieldError)
    className=""
    error="some error"
  />
</div>
`;
